src/SelfServe/Documentation/README.md

```diff
@@ -0,0 +1,197 @@
+
+#### Description
+This file will contain the actual code that is transalted into the UI component by the Self Serve framework.
+* Each Self Serve class
+  * Needs to extends the [SelfServeBase](../classes/selfserve_selfservetypes.selfservebaseclass.html) class.
+  * Needs to have the [@IsDisplayable()](./selfserve_decorators.html#isdisplayable) decorator to tell the compiler that UI needs to be generated from this class.
+  * Needs to define an [onSave()](../classes/selfserve_selfservetypes.selfservebaseclass.html#onsave) function, a callback for when the save button is clicked.
+  * Needs to define an [initialize()](../classes/selfserve_selfservetypes.selfservebaseclass.html#initialize) function, to set default values for the inputs.
+  * Needs to define an [onRefresh()](../classes/selfserve_selfservetypes.selfservebaseclass.html#onrefresh) function, a callback for when the refresh button is clicked.
+  * Can have an optional [@RefreshOptions()](./selfserve_decorators.html#refreshoptions) decorator that determines how often the auto refresh of the UI component should take place.
+
+* Each property of the Self Serve class
+  * Having a [@Values()](./selfserve_decorators.html#values) decorator will be translated into an UI element.
+  * Decorated with [@Values()](./selfserve_decorators.html#values) can have an optional [@PropertyInfo()](./selfserve_decorators.html#propertyinfo) decorator that describes it's info bubble.
+  * Can have an optional [@OnChange()](./selfserve_decorators.html#onchange) decorator that dictates the effects of the change of the UI element tied to this property.
+
+```ts
+@IsDisplayable()
+@RefreshOptions({ retryIntervalInMs: 2000 })
+export default class NewFeature extends SelfServeBaseClass {
+
+  public onRefresh = async (): Promise<RefreshResult> => {
+      // refresh RP call and processing logic
+  };
+
+  public onSave = async (
+    currentValues: Map<string, SmartUiInput>,
+    baselineValues: ReadonlyMap<string, SmartUiInput>
+  ): Promise<OnSaveResult> => {
+      // onSave RP call and processing logic
+  }
+
+  public initialize = async (): Promise<Map<string, SmartUiInput>> => {
+      // initialize RP call and processing logic
+  }
+
+  @Values(...)
+  stringProperty: string;
+
+  @OnChange(...)
+  @PropertyInfo(...)
+  @Values(...)
+  numberProperty: number;
+}
+```
+
+### 2. RP file
+
+#### Naming Convention
+SELF_SERVE_CLASS_NAME.rp.ts
+
+#### Example
+[SelfServeExample.rp.ts](https://github.com/Azure/cosmos-explorer/blob/master/src/SelfServe/Example/SelfServeExample.rp.ts)
+
+#### Description
+The RP file will host the REST calls needed for the save, initialize and refresh functions. This decouples the code that builds the UI and the code that makes REST calls. Inbuilt objects and methods are exposed to access the Azure Cosmos DB subscription id, resource group name and database account name that are needed to make the REST calls.
```

Comment:
>Inbuilt objects and methods are exposed to acce [](http://example.com/codeflow?start=171&length=47)

Explicitly say:

we can use usercontext object to get sub id, rg, acc name.

Use armrequestwithoutpolling to make the arm api call as shown in the below example.